src/app/modules/shared/services/shopping.cart.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { OrderProduct } from '../types/product.order';
+
+@Injectable({ providedIn: 'root' })
+export class ShoppingCartService {
+  private shoppingCart: OrderProduct[] = [];
+  public getShoppingCart() {
+    return this.shoppingCart;
+  }
+
+  public addToShoppingCart(product: OrderProduct): OrderProduct[] {
+    const foundProduct = this.shoppingCart.find((p) => p.id === product.id);
+    if (!foundProduct) {
+      product.quantity = 0;
+      product.quantity++;
+      this.shoppingCart.push(product);
+    } else {
+      this.shoppingCart.forEach((findProduct) => {
+        if (findProduct.id === product.id) {
+          findProduct.quantity++;
+        }
+      });
```

Comment:
Here you don't need to go through the entire list to find the product you need. Look for [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) method